patroni/log.py

```diff
@@ -271,14 +272,73 @@ def reload_config(self, config: Dict[str, Any]) -> None:
                     new_handler = logging.StreamHandler()
                 handler = new_handler or self.log_handler
 
+            oldlogtype = (self._config or {}).get('type', PatroniLogger.DEFAULT_TYPE)
+            logtype = config.get('type', PatroniLogger.DEFAULT_TYPE)
+
             oldlogformat = (self._config or {}).get('format', PatroniLogger.DEFAULT_FORMAT)
             logformat = config.get('format', PatroniLogger.DEFAULT_FORMAT)
 
             olddateformat = (self._config or {}).get('dateformat') or None
             dateformat = config.get('dateformat') or None  # Convert empty string to `None`
 
-            if (oldlogformat != logformat or olddateformat != dateformat or new_handler) and handler:
-                handler.setFormatter(logging.Formatter(logformat, dateformat))
+            old_static_fields = (self._config or {}).get('static_fields', {})
+            static_fields = config.get('static_fields', {})
+
+            if (
+                not deep_compare({oldlogtype: oldlogformat}, {logtype: logformat})
+                or olddateformat != dateformat
+                or deep_compare(old_static_fields, static_fields)
+                or new_handler
+            ) and handler:
+                if logtype == 'json':
+                    if isinstance(logformat, str):
+                        jsonformat = logformat
+                        rename_fields = {}
+                    elif isinstance(logformat, list):
+                        log_fields: List[str] = []
+                        rename_fields: Dict[str, str] = {}
+
+                        for field in logformat:
+                            if isinstance(field, str):
+                                log_fields.append(field)
+                            elif isinstance(field, dict):
+                                for original_field, renamed_field in field.items():
+                                    if isinstance(renamed_field, str):
+                                        log_fields.append(original_field)
+                                        rename_fields[original_field] = renamed_field
+                                    else:
+                                        _LOGGER.exception(
+                                            f'Expected renamed log field to be a string, but got "{field}"'
+                                        )
```

Comment:
For logging we use %-formatting, rather than f-strings or str.format.
Also, please downgrade exception to a warning.

Reply:
Even for `ImportError`? I agree with other logs but is not error better for this case?

Reply:
For ImportError it could be error of course, with some details printed using `%r` placeholder.